catalog/transformer/catalog-transformer-tika-input/src/main/java/ddf/catalog/transformer/input/tika/TikaInputTransformer.java

```diff
@@ -99,7 +99,163 @@
     private Map<ServiceReference, ContentMetadataExtractor> contentMetadataExtractors =
             Collections.synchronizedMap(new TreeMap<>(new ServiceComparator()));
 
-    private MetacardType metacardType = null;
+    private MetacardType fallbackExcelMetacardType = null;
+
+    private MetacardType fallbackJpegMetacardType = null;
+
+    private MetacardType fallbackMp4MetacardType = null;
+
+    private MetacardType fallbackMpegMetacardType = null;
+
+    private MetacardType fallbackOfficeDocMetacardType = null;
+
+    private MetacardType fallbackPdfMetacardType = null;
+
+    private MetacardType fallbackPowerpointMetacardType = null;
+
+    // commonTikaMetacardType represents the MetacardType to be used when an ingested product's mime
+    // type does not match a mime type that is supported by the mimeTypeToMetacardTypeMap
+    private MetacardType commonTikaMetacardType = null;
+
+    public void setCommonTikaMetacardType(MetacardType metacardType) {
+        this.commonTikaMetacardType = metacardType;
+    }
+
+    public void setFallbackExcelMetacardType(MetacardType metacardType) {
+        this.fallbackExcelMetacardType = metacardType;
+    }
+
+    public void setFallbackJpegMetacardType(MetacardType metacardType) {
+        this.fallbackJpegMetacardType = metacardType;
+    }
+
+    public void setFallbackMp4MetacardType(MetacardType metacardType) {
+        this.fallbackMp4MetacardType = metacardType;
+    }
+
+    public void setFallbackMpegMetacardType(MetacardType metacardType) {
+        this.fallbackMpegMetacardType = metacardType;
+    }
+
+    public void setFallbackOfficeDocMetacardType(MetacardType metacardType) {
+        this.fallbackOfficeDocMetacardType = metacardType;
+    }
+
+    public void setFallbackPdfMetacardType(MetacardType metacardType) {
+        this.fallbackPdfMetacardType = metacardType;
+    }
+
+    public void setFallbackPowerpointMetacardType(MetacardType metacardType) {
+        this.fallbackPowerpointMetacardType = metacardType;
+    }
+
+    private Map<String, MetacardType> mimeTypeToMetacardTypeMap = new HashMap<>();
+
+    /**
+     * Populates the mimeTypeToMetacardMap for use in determining the {@link MetacardType} that
+     * corresponds to an ingested product's mimeType.
+     */
+
+    public void populateMimeTypeMap() {
+        mimeTypeToMetacardTypeMap.put("application/vnd.ms-powerpoint.presentation.macroenabled.12",
+                fallbackPowerpointMetacardType);
+
+        mimeTypeToMetacardTypeMap.put("application/vnd.ms-powerpoint", fallbackPowerpointMetacardType);
```

Comment:
Thanks for the suggestion. I was able to use guava’s media type class for some of the strings but unfortunately a some of the strings I need don't exist in that class.